collector/aws/elasticips.go

```diff
@@ -0,0 +1,186 @@
+package aws
+
+import (
+	"finala/collector"
+	"finala/collector/config"
+	"fmt"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	log "github.com/sirupsen/logrus"
+)
+
+// ElasticIPClientDescreptor is an interface defining the aws ec2 client
+type ElasticIPClientDescreptor interface {
+	DescribeAddresses(input *ec2.DescribeAddressesInput) (*ec2.DescribeAddressesOutput, error)
+}
+
+// ElasticIPManager will hold the elastic ip manger strcut
+type ElasticIPManager struct {
+	collector          collector.CollectorDescriber
+	client             ElasticIPClientDescreptor
+	pricingClient      *PricingManager
+	metric             config.ResourceConfig
+	region             string
+	servicePricingCode string
+	rateCode           string
+	Name               string
+}
+
+// DetectedElasticIP define the detected AWS elastic ip
+type DetectedElasticIP struct {
+	Region        string
+	Metric        string
+	IP            string
+	PricePerHour  float64
+	PricePerMonth float64
+	Tag           map[string]string
+}
+
+// NewElasticIPManager implements AWS GO SDK
+func NewElasticIPManager(collector collector.CollectorDescriber, client ElasticIPClientDescreptor, pricing *PricingManager, metric config.ResourceConfig, region string) *ElasticIPManager {
+
+	return &ElasticIPManager{
+		collector:          collector,
+		client:             client,
+		pricingClient:      pricing,
+		metric:             metric,
+		region:             region,
+		servicePricingCode: "AmazonEC2",
+		rateCode:           "JTU8TKNAMW",
+		Name:               fmt.Sprintf("%s_elastic_ip", ResourcePrefix),
+	}
+}
+
+// Detect checks if elastic ips is under utilize
+func (ei *ElasticIPManager) Detect() ([]DetectedElasticIP, error) {
+
+	log.WithFields(log.Fields{
+		"region":   ei.region,
+		"resource": "elastic ips",
+	}).Info("starting to analyze resource")
+
+	ei.collector.UpdateServiceStatus(collector.EventCollector{
+		ResourceName: ei.Name,
+		Data: collector.EventStatusData{
+			Status: collector.EventFetch,
+		},
+	})
+
+	elasticIPs := []DetectedElasticIP{}
+
+	priceFIlters := ei.GetPricingFilterInput()
+	// Getting elastic ip pricing
+	price, err := ei.pricingClient.GetPrice(priceFIlters, ei.rateCode, ei.region)
+	if err != nil {
+		log.WithError(err).WithFields(log.Fields{
+			"rate_code":     ei.rateCode,
+			"region":        ei.region,
+			"price_filters": priceFIlters,
+		}).Error("could not get elastic ip price")
+		ei.updateErrorServiceStatus(err)
+		return elasticIPs, err
+	}
+
+	// Getting all elastic ip addressess
+	ips, err := ei.DescribeAddressess()
+	if err != nil {
+		ei.updateErrorServiceStatus(err)
+		return elasticIPs, err
+	}
+
+	for _, ip := range ips {
+
+		// Checks id the ip that are not associated with a running Amazon Elastic Compute Cloud (Amazon EC2) instance
```

Comment:
Can you please change this comment to be more clear I find it hard to understand it.

Reply:
removed